### PR TITLE
[WIP] use junit 5.9.2 in targets

### PR DESCRIPTION
**Pull request by cdietrich**
use junit 5.9.2 in targets

### CODE REVIEW

File: org.eclipse.xtext.tests/.classpath [hunk 1]
```diff
@@ -6,7 +6,6 @@
 		</attributes>
 	</classpathentry>
 	<classpathentry kind="con" path="org.eclipse.pde.core.requiredPlugins"/>
-	<classpathentry kind="con" path="org.eclipse.jdt.junit.JUNIT_CONTAINER/5"/>
```
Comment (cdietrich) [inline org.eclipse.xtext.tests/.classpath:9]: with the container tycho fails to resolve
without you have to select junit4 for the junit4 tests in run as dialogs. did not find a nice solution yet
see discussion in https://github.com/eclipse/xtext/issues/2197

Comment (cdietrich) [reply org.eclipse.xtext.tests/.classpath:9]: it is also about running single tests. currently in maintainance mode this is 50% of my usecases.
maybe there is also another option to make the vintage enginge visble to jdt.

Comment (cdietrich) [reply org.eclipse.xtext.tests/.classpath:9]: i dont know. 
the problem is we mix tests and a plugin.
and we need the imports for the plugin

Comment (LorenzoBettini) [reply org.eclipse.xtext.tests/.classpath:9]: What do you mean by "we need the imports for the plugin"?

Comment (cdietrich) [reply org.eclipse.xtext.tests/.classpath:9]: xtext.testing has tests and is a real plugin

Comment (LorenzoBettini) [reply org.eclipse.xtext.tests/.classpath:9]: Yes, I meant that: I seem to understand from the discussion that the JUnit container in the classpath should be enough (without import packages in the MANIFEST) to make everyone happy (running tests from Eclipse and Tycho).

Comment (cdietrich) [reply org.eclipse.xtext.tests/.classpath:9]: will it work if you use xtext.testing in mydsl.tests then?i assume no

Comment (cdietrich) [reply org.eclipse.xtext.tests/.classpath:9]: it is a plugin that is used in other plugins

Comment (cdietrich) [reply org.eclipse.xtext.tests/.classpath:9]: yes this is what i mean. you have to selct junit 4 in the dialog.

i have doubts a plugin can have code in osgi that is not on classpath. but i am not an osgi expert.
am also not sure how to properly test this.

**Comment by HannesWell** [comment]
The class org.eclipse.xtend.ide.buildpath.Junit5LibClasspathAdder also maintains a list of junit bundles that probably has to be adapted too.
I wonder if it would be possible to make that more dynamic and use the BSN of those bundles that contain the corresponding Junit classes?

**Comment by cdietrich** [comment]
> Junit5LibClasspathAdder

this is a good question. it all is a  mess and was even more with older eclipse and xtend versions.
wonder if we can use package imports there.
or a mix of both

**Comment by cdietrich** [comment]
@LorenzoBettini @HannesWell  any idea why this is not working

Bildschirmfoto 2023-04-13 um 12 50 16

**Comment by LorenzoBettini** [comment]
@cdietrich I can reproduce it, but I have no idea why that's not working... even because that bundle is in the target platform...

**Comment by cdietrich** [comment]
happens also in fresh eclipse.
created https://github.com/eclipse-pde/eclipse.pde/issues/574 to clarify

**Comment by cdietrich** [comment]
will close this one and create a rebased new on 2morrow

**Pull request closed without merging**